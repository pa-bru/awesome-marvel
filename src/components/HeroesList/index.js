import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Grid from '@material-ui/core/Grid'

import HeroCard from 'components/HeroCard'

class HeroesList extends Component {
  static propTypes = {
    heroes: PropTypes.array.isRequired,
  }

  static defaultProps = {
    heroes: [],
  }

  render() {
    const { heroes } = this.props
    return (
      <Grid container spacing={24}>
        {heroes.map((hero, index) => (
          <Grid item key={index} style={{ display: 'flex' }} xs={12} sm={6} md={4} lg={3}>
            <HeroCard {...hero} />
          </Grid>
        ))}
      </Grid>
    )
  }
}

export default HeroesList
