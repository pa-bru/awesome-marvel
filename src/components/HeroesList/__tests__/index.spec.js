import React from 'react'
import { shallow } from 'enzyme'
import HeroesList from '../'

describe('HeroCard', () => {
  it('should render correctly', () => {
    // When
    const tree = shallow(<HeroesList />)

    // Then
    expect(tree).toMatchSnapshot()
  })

  it('should render a list of heroes', () => {
    // Given
    const heroes = [
      {
        id: 1,
      },
      {
        id: 2,
      },
    ]

    // When
    const tree = shallow(<HeroesList heroes={heroes} />)

    // Then
    expect(tree).toMatchSnapshot()
  })
})
