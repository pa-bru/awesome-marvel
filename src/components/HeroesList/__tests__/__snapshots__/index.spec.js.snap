// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCard should render a list of heroes 1`] = `
<GridWrapper
  container={true}
  spacing={24}
>
  <GridWrapper
    item={true}
    key="0"
    lg={3}
    md={4}
    sm={6}
    style={
      Object {
        "display": "flex",
      }
    }
    xs={12}
  >
    <WithStyles(HeroCard)
      id={1}
    />
  </GridWrapper>
  <GridWrapper
    item={true}
    key="1"
    lg={3}
    md={4}
    sm={6}
    style={
      Object {
        "display": "flex",
      }
    }
    xs={12}
  >
    <WithStyles(HeroCard)
      id={2}
    />
  </GridWrapper>
</GridWrapper>
`;

exports[`HeroCard should render correctly 1`] = `
<GridWrapper
  container={true}
  spacing={24}
/>
`;
